packages/react-native-tracing/ios/NativeInstrumentation.swift

```diff
@@ -26,28 +15,59 @@ public class NativeInstrumentation: NSObject, RCTBridgeModule {
         return "NativeInstrumentation"
     }
     
-    @objc
-    public func getStartupTime(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
-        guard let startTime = NativeInstrumentation.startTime else {
-            reject("NO_START_TIME", "[NativeInstrumentation] Start time was not initialized", nil)
+    override init() {
+        super.init()
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(handleBundleLoadStart(_:)),
+            name: NSNotification.Name("RCTJavaScriptWillStartLoadingNotification"),
+            object: nil
+        )
+    }
+    
+    @objc private func handleBundleLoadStart(_ notification: Notification) {
+        if NativeInstrumentation.hasAppRestarted {
             return
         }
+
+        NativeInstrumentation.hasAppRestarted = true
+    }
+    
+    @objc
+    public func getStartupTime(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
```

Comment:
Not sure if anything can fail here, but I see you reject the promise with `STARTUP_TIME_ERROR` for Android, maybe to keep consistency across platforms you can do the same here

Reply:
Good catch!